src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -187,6 +188,90 @@ public void Clear()
             }
         }
 
+        /// <summary>
+        /// Atomically searches for a specified key in the table and returns the corresponding value. If the
+        /// key does not exist in the table, the method adds the given value and binds it to the specified key.
+        /// </summary>
+        /// <param name="key">The key of the value to find. It cannot be <see langword="null"/>.</param>
+        /// <param name="value">The value to add and bind to <typeparamref name="TKey"/>, if one does not exist already.</param>
+        /// <returns>The value bound to <typeparamref name="TKey"/> in the current <see cref="ConditionalWeakTable{TKey, TValue}"/> instance, after the method completes.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="key"/> is <see langword="null"/>.</exception>
+        public TValue GetOrAdd(TKey key, TValue value)
+        {
+            // key is validated by TryGetValue
+            if (TryGetValue(key, out TValue? existingValue))
+            {
+                return existingValue;
+            }
+
+            return GetOrAddLocked(key, value);
+        }
+
+        /// <summary>
+        /// Atomically searches for a specified key in the table and returns the corresponding value.
+        /// If the key does not exist in the table, the method invokes the supplied factory to create
+        /// a value that is bound to the specified key.
```

Comment:
The "atomically" here might be misleading, since the delegate is invoked without the lock held.

Reply:
Removed! All other methods don't use the word either anyway, and just assume they're all thread-safe.